doc/index.md

```diff
@@ -1,61 +1,119 @@
-local body = assert(stream:get_body_as_string())
-if headers:get ":status" ~= "200" then
-	error(body)
+            local response, err, errno = ws:receive()
+            if not response then
+                io.stderr:write(string.format("%s: %s %s %s\n", err, errno, "Receive failed", debug.traceback()))
+            else
+                print("Received: " .. response .. "- sizeof: " .. #response)
+            end
+        until shutdown == true
+    end)
+
+    assert(cq:loop())
+else
+    io.stderr:write(string.format("%s: %s %s %s\n", err, errno, "Failed to connect.", debug.traceback()))
 end
-print(body)
 ```
 
-
 ## Asynchronous Operation
 
-All lua-http operations including DNS lookup, socket connection, TLS negotiation and read/write operations are all asynchronous when run inside of a cqueue.
-[Cqueues](http://25thandclement.com/~william/projects/cqueues.html) is a lua library that allows for composable event loops.
-Cqueues can be integrated with almost any main loop or event library you may encounter (see [here](https://github.com/wahern/cqueues/wiki/Integrations-with-other-main-loops) for more information + samples), and hence lua-http can be asynchronous in any place you write lua!
+lua-http has been written to perform asyncrhounously to greatly reduce resource overhead in an application. Asynchronous operations are achieved by utilizing cqueues, a Lua/C library that incorporates Lua yielding and kernel level APIs to reduce CPU usage. Most lua-http operations including DNS lookup, TLS negotiation and read/write operations are asynchronous. While sometimes it is necessary to block a thread, any operations that may block the current coroutine take an optional timeout to ensure your application does not hang in an unusable state.
+
+Asynchrounous operations is one of the most powerful features of lua-http. For instance, an HTTP server can be instantiated within any Lua main loop and run along side application code without adversely affecting the main application process. If other cqueue enabled components are integrated within a cqueue loop, the application is entirely event driven through kernel level polling APIs. Likewise, an application can be made "internet enabled" by simply "wrapping" an HTTP or WebSocket client request within a main loop and processing responses.
+
+Cqueues can be used in conjunction with lua-http to integrate other features into your lua application and create powerful, performant, web enabled applications. Some of the examples in this guide will use cqueues for simple demonstrations. For more resources about cqueues, please see:
+
+- [The cqueues website](http://25thandclement.com/~william/projects/cqueues.html) for more information about library. Other resources:
+
+- Cqueue examples can be found with the cqueues source code available through [git or archives](http://www.25thandclement.com/~william/projects/cqueues.html#download). 
```

Comment:
perhaps easier to link to github examples directory? https://github.com/wahern/cqueues/tree/master/examples